### PR TITLE
Views Sort Plugin Generator

### PR DESCRIPTION
**Pull request by NickDickinsonWilde**

### CODE REVIEW

File: src/Command/Drupal_8/Plugin/Views/Sort.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+use DrupalCodeGenerator\Command\BaseGenerator;
+use DrupalCodeGenerator\Utils;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Implements d8:plugin:views:sort command.
+ */
+class Style extends BaseGenerator {
```
Comment (Chi-teck) [inline src/Command/Drupal_8/Plugin/Views/Sort.php]: The class name should _Sort_ I presume.

File: src/Command/Drupal_8/Plugin/Views/Sort.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  protected $name = 'd8:plugin:views:sort';
+  protected $description = 'Generates views sort plugin';
```
Comment (manuee) [inline src/Command/Drupal_8/Plugin/Views/Sort.php]:
```suggestion
  protected $description = 'Generates a views sort plugin';
```

File: templates/d8/plugin/views/sort-plugin.twig [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   */
+  protected $usesRowClass = TRUE;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    return $options;
+  }
```
Comment (manuee) [inline templates/d8/plugin/views/sort-plugin.twig]: Do we really need the defineOptions() method defined?

File: templates/d8/plugin/views/sort-plugin.twig [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * {{ plugin_label }} sort plugin.
+ *
+ * @ViewsSort("{{ plugin_id }}")
+ */
+class {{ class }} extends SortPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $usesRowPlugin = TRUE;
```
Comment (manuee) [inline templates/d8/plugin/views/sort-plugin.twig]: I  believe this is for sytle plugins only.

File: templates/d8/plugin/views/sort-plugin.twig [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $usesRowClass = TRUE;
```
Comment (manuee) [inline templates/d8/plugin/views/sort-plugin.twig]: I  believe this is for sytle plugins only.

File: templates/d8/plugin/views/sort-plugin.twig [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
```
Comment (manuee) [inline templates/d8/plugin/views/sort-plugin.twig]: This method is an exact copy of `SortPluginBase::buildOptionsForm()`, I dont think we need to add it to the generated plugin.

**Review by manuee — CHANGES_REQUESTED** [review]
Besides by previous comments, I believe the method most sort plugins will implement will be the `query(`) method. I propose we add it.

**Comment by Rahul-coder01** [comment]
https://wiki.almworks.com/display/structure/Sort+Generators

**Comment by Chi-teck** [comment]
Seems not many people interested in this generator. Closing.

**Pull request closed without merging**